src/helpers/getByAPI.js

```diff
@@ -100,7 +100,12 @@ export const getByText = (instance: ReactTestInstance) =>
     try {
       return instance.find((node) => getNodeByText(node, text));
     } catch (error) {
-      throw new ErrorWithStack(prepareErrorMessage(error), getByTextFn);
+      throw new ErrorWithStack(
+        `${prepareErrorMessage(error)} with text '${
+          typeof text === 'string' ? text : text.toString()
+        }'`,
```

Comment:
```suggestion
        `${prepareErrorMessage(error)} with text '${text}'`,
```

We should probably be able to the if condition, sa template string should do the checks & toString() call for us.

Reply:
That was my first idea but I was getting this Flow error:

![image](https://user-images.githubusercontent.com/1479283/94341906-552aa400-ffdb-11ea-9a41-dd8b63bd497d.png)

```
Cannot coerce `text` to string because  `RegExp` [1] should not be coerced.
```

Reply:
```javascript
${text.toString}
```
seems to work
![image](https://user-images.githubusercontent.com/1479283/94341929-830fe880-ffdb-11ea-8567-235cd252c551.png)

Would you be happy with that?

Reply:
@sregg doing `text.toString()` is the way to go here, yea 👍

Reply:
Btw, I think that `prepareErrorMessage` function is a perfect place for handling this extra text. Similar to what you have in a test here: https://github.com/callstack/react-native-testing-library/pull/550/files#diff-4246008ab708750218ff9e41118401a1R14